tests/test_data/test_address.py

```diff
@@ -10,151 +10,220 @@
+        current_locale = address.get_current_locale()
+
+        if current_locale in p.POSTAL_CODE_REGEX:
+            assert re.match(p.POSTAL_CODE_REGEX[current_locale], result)
+        else:
+            assert re.match(p.POSTAL_CODE_REGEX['default'], result)
+
+    def test_country(self, address):
+        result = address.country()
+        assert result in address._data['country']['name']
+
+    @pytest.mark.parametrize(
+        'fmt, length', [
+            (CountryCode.ISO2, 2),
+            (CountryCode.ISO3, 3),
+            (CountryCode.NUMERIC, 3),
+            (None, [2, 3]),
+        ],
+    )
+    def test_country_iso(self, _address, fmt, length):
+        iso = _address.country_iso_code(fmt=fmt)
+
+        if fmt is not None:
+            assert iso in COUNTRIES_ISO[fmt.value]
+
+        assert len(iso) == length or len(iso) in length
+
+        with pytest.raises(NonEnumerableError):
+            _address.country_iso_code(fmt='nil')
+
+    def test_city(self, address):
+        result = address.city()
+        assert result in address._data['city']
+
+    def test_continent(self, address):
+        result = address.continent()
+        assert result in address._data['continent']
+
+        result = address.continent(code=True)
+        assert result in CONTINENT_CODES
+
+    def test_calling_code(self, _address):
+        result = _address.calling_code()
+        assert result is not None
+        assert result in CALLING_CODES
+
+
+class TestSeededAddress(object):
+    TIMES = 5
+
+    @pytest.fixture
+    def _addresses(self):
+        return Address(seed=seed), Address(seed=seed)
+
+    def test_street_number(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.street_number() == a2.street_number()
+            assert a1.street_number(maximum=42) == a2.street_number(maximum=42)
+
+    def test_latitude(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.latitude() == a2.latitude()
+
+    def test_longitude(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.longitude() == a2.longitude()
+
+    def test_coordinates(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.coordinates() == a2.coordinates()
+
+    def test_street_name(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.street_name() == a2.street_name()
+
+    def test_street_suffix(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.street_suffix() == a2.street_suffix()
+
+    def test_address(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.address() == a2.address()
+
+    def test_state(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
```

Comment:
Are there no other much elegant way to repeat test `n` time? Something like:
```
@pytest.mark.repeat(self.TIMES)
def test_state(self, _addresses):
    a1, a2 = _addresses
    assert a1.state() == a2.state()
    assert a1.state(abbr=True) == a2.state(abbr=True)
```

Reply:
Why do even need to repeat a test?

Reply:
I think to ensure that result is same for each iteration.

Reply:
@lk-geimfari If I understand `pytest` correctly, when we writing tests with decorator, providers initialization at runtime will be at every iteration.

Reply:
@duckyou You're right. My idea was not really correct or rather really incorrect.